core/src/scene/sceneLayer.cpp

```diff
@@ -1,70 +1,100 @@
 #include "sceneLayer.h"
 
 #include <algorithm>
+#include <queue>
 
 namespace Tangram {
 
-SceneLayer::SceneLayer(std::string _name, Filter _filter, std::vector<DrawRule> _rules,
+SceneLayer::SceneLayer(std::string _name, Filter _filter,
+                       std::vector<StaticDrawRule> _rules,
                        std::vector<SceneLayer> _sublayers) :
     m_filter(_filter),
     m_name(_name),
     m_rules(_rules),
-    m_sublayers(_sublayers),
-    m_depth(0) {
+    m_sublayers(_sublayers) {
 
-    // Rules must be sorted to merge correctly
+    // Rules must be sorted to merge correctly - not anymore
     std::sort(m_rules.begin(), m_rules.end());
 
-    // m_depth is one more than the maximum depth of any sublayer
-    for (auto& sublayer : m_sublayers) {
-        m_depth = sublayer.m_depth > m_depth ? sublayer.m_depth : m_depth;
-    }
-    m_depth++;
+}
+
+void addStyleParams(const std::vector<StaticDrawRule>& rules, Styling& _styling){
+    for (auto& rule : rules) {
+        auto& map = _styling.styles;
 
-    // Sublayers must be sorted by the depth of their deepest leaf in order to
-    // correctly traverse them in match()
-    std::sort(m_sublayers.begin(), m_sublayers.end(),
-              [&](const auto& a, const auto& b) {
-                  return a.m_depth > b.m_depth;
-              });
+        auto it = std::find_if(map.begin(), map.end(), [&](auto& m){
```

Comment:
In Release builds - not default builds, and was the test with integers? 

I thought of changing this part by using pre-allocated DrawRules for each style name so that one has a direct lookup.